MSAL/test/app/mac/MSALCacheViewController.m

```diff
@@ -26,16 +26,305 @@
+        [self loadCache];
+    }
+}
+
+- (void)deleteToken:(MSIDBaseToken *)token
+{
+    if (token)
+    {
+        switch (token.credentialType)
+        {
+            case MSIDRefreshTokenType:
+            {
+                [self.defaultAccessor validateAndRemoveRefreshToken:(MSIDRefreshToken *)token
+                                                            context:nil
+                                                              error:nil];
+                break;
+            }
+            default:
+                [self.defaultAccessor removeToken:token context:nil error:nil];
+                break;
+        }
+        
+        [self loadCache];
+    }
+}
+
+#pragma mark Button Actions
+
+- (IBAction)refreshCache:(id)sender
+{
+    [self loadCache];
+}
+
+
+- (IBAction)expireOrInvalidateToken:(id)sender
+{
+    id item = [self.outLineView itemAtRow:[self.outLineView selectedRow]];
+    
+    if ([item isKindOfClass:[MSIDBaseToken class]])
+    {
+        if ([item isKindOfClass:[MSIDRefreshToken class]])
+        {
+            [self invalidateRefreshToken:(MSIDRefreshToken *)item];
+        }
+        else if([item isKindOfClass:[MSIDAccessToken class]])
+        {
+            [self expireAccessToken:(MSIDAccessToken *)item];
+        }
+    }
+}
+
+- (IBAction)deleteItem:(id)sender
+{
+    id item = [self.outLineView itemAtRow:[self.outLineView selectedRow]];
+    if ([item isKindOfClass:[MSIDAccount class]])
+    {
+        [self deleteAllEntriesForAccount:(MSIDAccount *)item];
+    }
+    else if ([item isKindOfClass:[MSIDBaseToken class]])
+    {
+        [self deleteToken:(MSIDBaseToken *)item];
+    }
+    else if ([item isKindOfClass:[MSIDAppMetadataCacheItem class]])
+    {
+        [self deleteAppMetadata:(MSIDAppMetadataCacheItem *)item];
+    }
+}
+
+#pragma mark NSOutlineView Data Source Test Methods
+
+- (NSInteger)outlineView:(NSOutlineView *)outlineView numberOfChildrenOfItem:(id)item
+{
+    return !item ? [[self.cacheDict allKeys] count] : [[self.cacheDict objectForKey:item] count];
+}
+
+- (BOOL)outlineView:(NSOutlineView *)outlineView isItemExpandable:(id)item
+{
+    return !item ? YES : [[self.cacheDict objectForKey:item] count];
+}
+
+- (id)outlineView:(NSOutlineView *)outlineView child:(NSInteger)index ofItem:(id)item
+{
+    return !item ? [[self.cacheDict allKeys] objectAtIndex:index] : [[self.cacheDict objectForKey:item] objectAtIndex:index];
+}
+
+- (NSView *)outlineView:(NSOutlineView *)outlineView viewForTableColumn:(NSTableColumn *)tableColumn item:(id)item
+{
+    NSString *identifier = [tableColumn identifier];
+    if ([identifier isEqualToString:@"cacheColumn"])
+    {
+        NSString *textValue;
+        NSTableCellView *cellView = [outlineView makeViewWithIdentifier:@"cacheCell" owner:self];
+        
+        id parent = [outlineView parentForItem:item];
+        if (!parent)
+        {
+            textValue = [self getUPN:item] ? [self getUPN:item] : s_appMetadata;
+        }
+        
```

Comment:
shall we remove this line for consistent code style? :)

Reply:
Addressed.